### ISSUE TITLE
Deadlock while syncing + JSONRPC

### ISSUE THREAD

**Issue by gavofyork** [ethcore/parity#1058]
syncing hanged when serving jsonrpc call while syncing

 ```[ #869247 f7e1…3e7e ]---[ 63 blk/s | 171 tx/s | 62835590 gas/s  //··· 0/4 peers, #899522, 0+30253 queued ···// mem: 0 bytes db, 213 KiB chain, 23 MiB queue, 302 KiB sync ]
[ #869387 9757…4099 ]---[ 28 blk/s | 167 tx/s | 40392121 gas/s  //··· 0/4 peers, #899522, 0+30125 queued ···// mem: 0 bytes db, 126 KiB chain, 23 MiB queue, 302 KiB sync ]
[ #869387 9757…4099 ]---[ 0 blk/s | 0 tx/s | 0 gas/s  //··· 0/4 peers, #899522, 0+30125 queued ···// mem: 0 bytes db, 126 KiB chain, 23 MiB queue, 302 KiB sync ]
[ #869387 9757…4099 ]---[ 0 blk/s | 0 tx/s | 0 gas/s  //··· 0/5 peers, #899522, 0+30125 queued ···// mem: 0 bytes db, 126 KiB chain, 23 MiB queue, 302 KiB sync ]
[ #869387 9757…4099 ]---[ 0 blk/s | 0 tx/s | 0 gas/s  //··· 0/5 peers, #899522, 0+30125 queued ···// mem: 0 bytes db, 126 KiB chain, 23 MiB queue, 302 KiB sync ]
...
```

was running `web3.eth.getBlock`, `web3.eth.getUncleCountByBlockNumber`, `web3.eth.getUncleByBlockNumberAndIndex` while syncing.

**Comment by sammy007** [comment]
While syncing several times from scratch I noticed same behaviour when not using jsonrpc.

**Comment by gavofyork** [comment]
possibly fixed by #1095
